.pre-commit-config.yaml

```diff
@@ -14,22 +14,13 @@ repos:
       - id: absolufy-imports
         name: absolufy-imports
         files: ^xarray/
-    # This wants to go before isort & flake8
-  - repo: https://github.com/PyCQA/autoflake
-    rev: "v2.0.0"
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    # Ruff version.
+    rev: 'v0.0.226'
     hooks:
-      - id: autoflake # isort should run before black as black sometimes tweaks the isort output
-        args: ["--in-place", "--ignore-init-module-imports"]
-  - repo: https://github.com/PyCQA/isort
-    rev: 5.11.4
-    hooks:
-      - id: isort
-  - repo: https://github.com/asottile/pyupgrade
-    rev: v3.3.1
-    hooks:
-      - id: pyupgrade
-        args:
-          - "--py38-plus"
```

Comment:
It doesn't look like this is supported by Ruff. We'll need to walk back the type changes for py38...

Reply:
I think we do support this! Unless I'm misunderstanding. But we use something closer to Black's API, like:

```py
[tool.ruff]
# Always generate Python 3.7-compatible code.
target-version = "py38"
```